source/slang/hlsl.meta.slang

```diff
@@ -8902,48 +8902,57 @@ ${{{{
 }}}}
 
 ${{{{
-struct AtomicOperationInfo
+struct InternalAtomicOperationInfo
 {
     const char* slangSuffix;
     const char* cudaSuffix;
     const char* glslSuffix;
     const char* hlslSuffix;
     const char* metalSuffix;
+    const char* spirvFloatSuffix;
+    const char* spirvUIntSuffix;
+    const char* spirvIntSuffix;
+
+    const char* assertExpr;
 };
 
-AtomicOperationInfo atomicOperationInfo[6] = {
-        { "add", "Add", "Add", "Add", "add" },
-        { "and", "And", "And", "And", "and" },
-        { "max", "Max", "Max", "Max", "max" },
-        { "min", "Min", "Min", "Min", "min" },
-        { "or", "Or", "Or", "Or", "or" },
-        { "xor", "Xor", "Xor", "Xor", "xor" }
-    };
+InternalAtomicOperationInfo internalAtomicOperationInfo[7] = {
+    { "add", "Add", "Add", "Add", "fetch_add", "FAddEXT", "IAdd", "IAdd", "true" },
+    { "and", "And", "And", "And", "fetch_and", "And", "And", "And", "!__isFloat<T>()" },
+    { "max", "Max", "Max", "Max", "fetch_max", "FMaxEXT", "UMax", "SMax", "true" },
+    { "min", "Min", "Min", "Min", "fetch_min", "FMinEXT", "UMin", "SMin", "true" },
+    { "or", "Or", "Or", "Or", "fetch_or", "Or", "Or", "Or", "!__isFloat<T>()" },
+    { "xor", "Xor", "Xor", "Xor", "fetch_xor", "Xor", "Xor", "Xor", "!__isFloat<T>()" },
+    { "exchange", "Exch", "Exchange", "Exchange", "exchange", "Exchange", "Exchange", "Exchange", "!__isFloat<T>()" },
+};
 
-for (AtomicOperationInfo atomicOp : atomicOperationInfo)
+for (InternalAtomicOperationInfo atomicOp : internalAtomicOperationInfo)
```

Comment:
Add a comment before this for loop to list all full function names we are generating, so it is easy to find the code from an intrinsic name, e.g. finding "InterlockedAdd" can lead us to this loop.